include/nil/crypto3/zk/commitments/batched_commitment.hpp

```diff
@@ -0,0 +1,246 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//---------------------------------------------------------------------------//
+
+#ifndef CRYPTO3_ZK_STUB_PLACEHOLDER_COMMITMENT_SCHEME_HPP
+#define CRYPTO3_ZK_STUB_PLACEHOLDER_COMMITMENT_SCHEME_HPP
+
+#include <set>
+#include <map>
+
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+#include <nil/crypto3/math/polynomial/polynomial_dfs.hpp>
+
+#include <nil/crypto3/zk/transcript/fiat_shamir.hpp>
+#include <nil/crypto3/zk/commitments/type_traits.hpp>
+#include <nil/crypto3/zk/commitments/detail/polynomial/eval_storage.hpp>
+
+namespace nil {
+    namespace crypto3 {
+        namespace zk {
+            namespace commitments {
+
+                template<typename FieldType, typename CommitmentType> 
+                struct commitment_scheme_params_type{
+                    using commitment_type = CommitmentType;
+                    using field_type = FieldType;
+                };
+
+                // Placeholder commitment scheme works with polynomial_dfs
+                template<typename ParamsType, typename TranscriptType, typename PolynomialType = typename math::polynomial_dfs<typename ParamsType::field_type::value_type>> 
+                class polys_evaluator{
+                public:
+                    using params_type = ParamsType;
+                    using commitment_type = typename ParamsType::commitment_type;
+                    using field_type = typename ParamsType::field_type;
+                    using transcript_type = TranscriptType;
+                    using poly_type = PolynomialType;
+
+                    eval_storage<field_type> _z;
+                    polys_evaluator(){}
+                protected:
+                    std::map<std::size_t, std::vector<poly_type>> _polys;
+                    std::map<std::size_t, bool> _locked; // _locked[batch] is true after it is commited
+                    std::map<std::size_t, std::vector<std::vector<typename field_type::value_type>>> _points;
+                protected:
+                    math::polynomial<typename field_type::value_type> get_V(const std::vector<typename field_type::value_type> &points) const{
+                        math::polynomial<typename field_type::value_type> V = {1};
+                        for( std::size_t xi_index = 0; xi_index < points.size(); xi_index++ ){
+                            V = V * math::polynomial<typename field_type::value_type>({-points[xi_index], 1});
```

Comment:
Is this the same, as multiplying the values in points, and only at the end create the polynomial?

Reply:
It's polynomial V(X) = (X-xi0)...(X-xi_k) where k is small (<=3).
I think, it's faster to process it as math::polynomial.